cmd/minikube/cmd/start.go

```diff
@@ -359,6 +370,22 @@ func provisionWithDriver(cmd *cobra.Command, ds registry.DriverState, existing *
 	}, nil
 }
 
+func validateKicBaseVersion(r command.Runner) {
+	res, err := r.RunCmd(exec.Command("cat", "/version.json"))
+	if err != nil {
+		out.WarningT("Unable to open version.json: {{.error}}", out.V{"error": err})
+	}
+
+	var versionDetails versionJSON
+	if err := json.Unmarshal(res.Stdout.Bytes(), &versionDetails); err != nil {
+		out.WarningT("Unable to parse version.json: {{.error}}", out.V{"error": err})
+	}
+
+	if versionDetails.MinikubeVersion != version.GetVersion() {
+		out.WarningT("Image was not built for the current minikube version: expected minikube version: {{.imageMinikubeVersion}} -> actual minikube version: {{.minikubeVersion}}", out.V{"imageMinikubeVersion": versionDetails.MinikubeVersion, "minikubeVersion": version.GetVersion()})
```

Comment:
We should have a solution message included so the user knows how to resolve this. Maybe `We recommend deleting your cluster and starting it again with the updated binary.`